src/pool/srv_pool.c

```diff
@@ -4740,7 +4740,7 @@ ds_pool_disconnect_handler(crt_rpc_t *rpc)
 {
 	uint8_t rpc_ver = opc_get_rpc_ver(rpc->cr_opc);
 
-	D_ASSERT(rpc_ver == DAOS_POOL_VERSION);
+	D_ASSERT(rpc_ver == DAOS_POOL_VERSION || rpc_ver == DAOS_POOL_VERSION - 1);
```

Comment:
I agree that we need to handle the cases of both `DAOS_POOL_VERSION` and `DAOS_POOL_VERSION - 1`. But does it means we must assertion for other cases? I am afraid that it will leave a security hole for the attack RPC from uncontrollable network, maybe malicious client/user.

Reply:
Yeah, let's remove the assertion and let following function handle it properly.